components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/summaryrow/SummaryRowView.kt

```diff
@@ -0,0 +1,136 @@
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.semantics.focused
+import androidx.compose.ui.semantics.role
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.style.TextOverflow
+import uk.gov.hmrc.components.compose.R
+import uk.gov.hmrc.components.compose.ui.theme.HmrcRippleTheme
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme.dimensions
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme.typography
+
+@Composable
+fun SummaryRowView(
+    titleText: String,
+    modifier: Modifier = Modifier,
+    isBoldTitleTextAppearance: Boolean = true,
+    titleMaxLines: Int = -1,
+    icon: Painter = painterResource(id = R.drawable.components_ic_chevron_right),
+    chevronContentDescription: String = "",
+    readerTrait: ReaderTrait = ReaderTrait.READER_TRAIT_INFO,
+    rows: List<@Composable () -> Unit>,
+    onSummaryRowClicked: (() -> Unit)? = null
+) {
+    if (onSummaryRowClicked != null) {
+        CompositionLocalProvider(LocalRippleTheme provides HmrcRippleTheme) {
+            Row(
+                modifier = modifier.clickable { onSummaryRowClicked.invoke() },
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                SummaryRow(
+                    modifier = Modifier
+                        .padding(
+                            top = dimensions.hmrcSpacing16,
+                            bottom = dimensions.hmrcSpacing16,
+                            start = dimensions.hmrcSpacing16
+                        )
+                        .weight(1f),
+                    titleText = titleText,
+                    titleMaxLines = titleMaxLines,
+                    isBoldTitleTextAppearance = isBoldTitleTextAppearance,
+                    readerTrait = readerTrait,
+                    rows = rows
+                )
+                Spacer(modifier = Modifier.width(dimensions.hmrcSpacing8))
+                Image(
+                    painter = icon,
+                    modifier = Modifier
+                        .semantics { role = Role.Button }
+                        .clickable { onSummaryRowClicked() },
+                    contentDescription = chevronContentDescription
+                )
+                Spacer(modifier = Modifier.width(dimensions.hmrcSpacing16))
+            }
+        }
+    } else {
+        SummaryRow(
+            modifier = modifier.padding(dimensions.hmrcSpacing16),
+            titleText = titleText,
+            titleMaxLines = titleMaxLines,
+            isBoldTitleTextAppearance = isBoldTitleTextAppearance,
+            readerTrait = readerTrait,
+            rows = rows
+        )
+    }
+}
+
+@Composable
+private fun SummaryRow(
+    modifier: Modifier = Modifier,
+    titleText: String,
+    isBoldTitleTextAppearance: Boolean,
+    titleMaxLines: Int,
+    readerTrait: ReaderTrait,
+    rows: List<@Composable () -> Unit>
```

Comment:
I know this is just private for use inside our library so it's not so important, but ideally this would follow the parameter ordering guidelines too for consistency and best practice
```suggestion
private fun SummaryRow(
    titleText: String,
    isBoldTitleTextAppearance: Boolean,
    titleMaxLines: Int,
    readerTrait: ReaderTrait,
    rows: List<@Composable () -> Unit>,
    modifier: Modifier = Modifier,
```